flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestQueue.java

```diff
@@ -119,6 +141,22 @@ private void enqueueAvailableReader(final NetworkSequenceViewReader reader) thro
 		}
 	}
 
+	private void flushReaders(long flushTimeout) throws Exception {
+		List<NetworkSequenceViewReader> readersToFlush = periodicFlushes.getReaders(flushTimeout);
+
+		boolean wasEmpty = availableReaders.isEmpty();
+
+		for (NetworkSequenceViewReader reader : readersToFlush) {
+			if (!reader.isRegisteredAsAvailable() && !reader.isBlocked()) {
```

Comment:
I think the flush operation indicates if this reader has both unfinished `BufferConsumer` and credits, we still want to transport this buffer to reduce latency. So the conditions of available reader should cover both available buffers and available credits.  But the current conditions only confirm the reader has available credits to insert into queue. When we poll this reader from the queue and get next buffer to find null, it seems not make sense to register available reader here.

So is it reasonable to adjust the conditions here to confirm this reader has both credits and unfinished buffers?